lib/kino/hub.ex

```diff
@@ -0,0 +1,33 @@
+defmodule Kino.Hub do
+  @moduledoc """
+  Functions related to hub integrations and Livebook apps.
+  """
+
+  @type app_info ::
+          %{type: :single}
+          | %{:type => :multi, optional(:creator) => user_info()}
+          | %{type: :none}
+
+  @type user_info :: %{
+          id: String.t(),
+          name: String.t() | nil,
+          email: String.t() | nil,
+          source: atom()
```

Comment:
I didn't make it more specific, because then it would be invalidated every time we add a new provider on Livebook side.